### PR TITLE
Deprecate imp module, part 2/2

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
As stated in the [first part of this change](https://github.com/Arelle/Arelle/pull/226), when importing and running Arelle as a third-party module, the following error is raised due to Python's `imp` module being deprecated:
```
arelle/PluginManager.py:11
  /workspaces/Arelle/arelle/PluginManager.py:11: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import os, sys, types, time, ast, imp, io, json, gettext, traceback
```

This PR removes the use of `imp` in favor of the currently supported `importlib` building on the method explained [here](https://docs.python.org/3/library/importlib.html#checking-if-a-module-can-be-imported).

#### Description of change
In this PR, I have replaced `imp.find_module` and `imp.find_module` with a helper function using importlib methods.

#### Steps to Test
We previously added a test to assert that `loadModule` properly loads a module.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by ggravlingen** [comment]
While our previosly setup unit test case runs, I'm slightly worried that we break things with this as I can't run the full test suite locally. Are plugins (eg the validators) used when running the full test suite here on Github?

File: arelle/PluginManager.py [hunk 1]
```diff
@@ -357,6 +362,25 @@ def _get_name_dir_prefix(
 
     return (None, None, None)
 
+def _find_and_load_module(moduleName: str, moduleDir: str) -> ModuleType | None:
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:365]: I've never considered that before. Would alphabetical make more sense?

File: arelle/PluginManager.py [hunk 2]
```diff
@@ -8,12 +8,16 @@
 
 '''
 from __future__ import annotations
-import os, sys, types, time, ast, imp, io, json, gettext, traceback
+import os, sys, types, time, ast, importlib, io, json, gettext, traceback
```
[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:11]: Great, thanks.

[the remark this answers is not in the document]
Comment (austinmatherne-wk) [reply arelle/PluginManager.py:11]: Looks like [there's an issue](https://github.com/Arelle/Arelle/runs/7369547242?check_suite_focus=true) loading plugins [by directory](https://github.com/Arelle/Arelle/blob/76cd68260a406c82cdb1fa96c6d1dd26f754b16a/tests/integration_tests/validation/ESEF/test_esef_ixbrl_conformance_suite.py#L9)

File: arelle/PluginManager.py [hunk 3]
```diff
@@ -393,7 +393,9 @@ def loadModule(moduleInfo: dict[str, Any], packagePrefix: str="") -> None:
         packagePrefix=packagePrefix,
     )
 
-    if moduleName is not None and moduleDir is not None and packageImportPrefix is not None:
+    if moduleName is None and moduleDir is None and packageImportPrefix is None:
```
[the remark this answers is not in the document]
Comment (sagesmith-wf) [reply arelle/PluginManager.py:396]: Nit perhaps a bit more pythonic do this instead: 

```python
all(p is None for p in [moduleName, moduleDir, packageImportPrefix])
```

but don't feel strongly

[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:396]: > 

I intentionally used `and` as `_get_name_dir_prefix` either returns a tuple of three `str` or a tuple of three `None`. But I don't have strong opinions here so I'll change it to the suggestion of @sagesmith-wf.

File: arelle/PluginManager.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (austinmatherne-wk) [reply arelle/PluginManager.py:11]: That test script is designed to execute against a conformance suite. I expect you'd get the same error running the command directly against a filing (`python -m arelleCmdLine --disclosureSystem esef --plugins validate/ESEF --validate --file <your-file>`), so I'd try that first.

Also, the ESEF conformance suite [is public](https://www.esma.europa.eu/document/conformance-suite-2021) if you'd like to run it locally.

File: arelle/PluginManager.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sagesmith-wf) [reply arelle/PluginManager.py]: I'm not attached either, I was simply informing what a "more pythonic" if statement might look like, but not pushing for one way or the other. Sorry for the confusion

File: arelle/PluginManager.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:11]: Thanks, I'll pull the conformance suite and test against it!

[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:11]: @austinmatherne-wk there is a lot of text output in the failing conformance tests but I assume the tests are actually failing due to the change made here?

[the remark this answers is not in the document]
Comment (austinmatherne-wk) [reply arelle/PluginManager.py:11]: Yes. I haven't looked deeply, but it appears that the validations from the plugin aren't executing.

[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:11]: OK, thanks, I'll keep digging into why.

[the remark this answers is not in the document]
Comment (ggravlingen) [reply arelle/PluginManager.py:11]: The latest push makes the XHTML-tests pass. My laptop might be a bit to slow to run the full test suite for the other files so let's see if they pass here.

File: arelle/PluginManager.py [hunk 4]
```diff
@@ -357,6 +362,29 @@ def _get_name_dir_prefix(
 
     return (None, None, None)
 
+def _get_location(moduleDir: str, moduleName: str) -> str:
+    """Get the file name of a plugin."""
+    if os.path.isfile(Path(f"{moduleDir}/{moduleName}.py")):
+        return Path(f"{moduleDir}/{moduleName}.py")
```
Comment (austinmatherne-wk) [inline arelle/PluginManager.py:368]: nit: define this path once as a variable and reuse it between the `return` statement and `isfile` conditional.

**Review by austinmatherne-wk — DISMISSED** [review]
Nice work @ggravlingen! This will require some manual testing on our part to verify the other plugins are still working properly, but it looks good. Thank you!  🎉

**Comment by ggravlingen** [comment]
> Nice work @ggravlingen! This will require some manual testing on our part to verify the other plugins are still working properly, but it looks good. Thank you!  🎉 

Thanks for considering the change! Please let me know if you run into any issues.

File: arelle/PluginManager.py (hunk 4 above)
Comment (ggravlingen) [reply arelle/PluginManager.py]: Just pushed this change

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Pull request merged**